File: 프로그래머스_코틀린/level2/카펫.kt
package 프로그래머스_코틀린.level2

class Solution6 {
    fun solution(brown: Int, yellow: Int): IntArray {
        var sum = brown / 2 + 2
        var multiply = yellow - 4 + 2 * sum

        for (i in 1..sum/2){
            if (i * (sum-i) == multiply){
                return intArrayOf(sum-i, i)
            }
        }
        return intArrayOf()
    }
}

//(a, b) (a>=b)
//brown = (a + b - 2) * 2
//yellow = (a-2) * (b-2)
//       = ab -2 (a+b) + 4

File: 프로그래머스/2/42842. 카펫/카펫.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int brown, int yellow) {
    int xy_sum = (brown + 4) / 2;
    int xy_multiply = yellow - 4 + 2 * xy_sum;
    
    for (int i = 0; i < xy_sum; i++){
        if (i * (xy_sum - i) == xy_multiply){
            return {xy_sum - i, i};
        }
    }
}

// # x+y = (brown+4) / 2
// # xy = yellow - 4 + 2(x+y)